docs/configuration.md

```diff
@@ -2801,6 +2801,32 @@ for {
 val z = (insertData *> readDatabase(id))
 ```
 
+#### `RedundantParens`: infix operator precedence
```

Comment:
Do we need this new setting? Shouldn't we just remove all the parens that are ok to removed based on the Scala operator precedence?

This looks like super complex setting :thinking:

Reply:
originally, we didn't have it, but a lot of people, along with the breaking changes, complained about too aggressive parens removal which interferes with readability (earlier implementation didn't detect removable infix parens). so, either we don't remove these parens as before, or provide a readability-compatible parameter...

the idea here is that removing parens around `a * b` is easy if followed by `... + c` (or preceded by `val foo = ...`), but not if followed by ` & d`.

Reply:
however, while we are figuring out the easiest way to deal with readability, let me split this change off since it's not technically a bug, like the others...

Reply:
ok, let me work on that. for now, i'll split it off so you can review just the bug fixes.

Reply:
@tgodzik done, will submit the infix change in a separate pr.